src/web/Clipboard.ts

```diff
@@ -6,16 +6,17 @@
 *
 * Web-specific implementation of the cross-platform Clipboard abstraction.
 */
+import escape = require('lodash/escape');
 
 import RX = require('../common/Interfaces');
 import SyncTasks = require('synctasks');
 
 export class Clipboard extends RX.Clipboard {
     public setText(text: string) {
         let node = Clipboard._createInvisibleNode();
-        // Replace carriage return /r with /r/n, so that pasting outside browser environment
+        // Replace carriage returns or newlines with br, so that pasting outside browser environment
         // (eg in a native app) preserves this new line
-        text = text.replace(/\r/g, '\r\n');
+        text = escape(text).replace(/\r\n?|\n/g, '<br />');
```

Comment:
isn't `/\r\n?|\n/g` equal with `/\r|\n/g`?

Reply:
I am not regexp expert, but IMO it is not. I think that /\r|\n/g won't cover CLRF case, but CL or RF separately.

Reply:
Totally agree. Already talked with @mshoho.